console-framework-client/src/main/java/io/axoniq/console/framework/messaging/HandlerMetricsRegistry.kt

```diff
@@ -43,36 +47,41 @@ class HandlerMetricsRegistry(
 
     private val noHandlerIdentifier = HandlerStatisticsMetricIdentifier(HandlerType.Origin, "application", MessageIdentifier("Dispatcher", componentName))
 
-    override fun registerLifecycleHandlers(lifecycle: Lifecycle.LifecycleRegistry) {
-        lifecycle.onStart(Phase.INSTRUCTION_COMPONENTS, this::start)
-    }
-
-    companion object {
-        private var instance: HandlerMetricsRegistry? = null
-        private var started: Boolean = false
-
-        /**
-         * Gets the current instance
-         */
-        fun getInstance() = instance
-    }
-
-    fun start() {
-        if (instance != null) {
-            logger.debug("HandlerMetricRegistry instance already started. Skipping new.")
-            return
+    init {
+        if(instance != null) {
+            logger.warn("HandlerMetricsRegistry already initialized. The new one will be the active.")
+            // Clear it to be sure. This is a situation that should not happen though.
+            instance?.onDisconnected()
         }
         instance = this
-        executor.scheduleAtFixedRate({
+        clientSettingsService.subscribeToSettings(this)
+    }
+
+    override fun onConnectedWithSettings(settings: ClientSettings) {
```

Comment:
It seems to me like `onConnectedWithSettings` now replaces the `start` we had in these components. It it clearer to call the start of a service `onConnectedWithSettings`, though?

Reply:
Although it doesn't only hold the start power, but also the restart when the settings update.

Reply:
Three-comments-in-short, I think another method name for `onConnectedWithSettings` would make the flow clearer from the perspective of a `ClientSettingsObserver` implementation.

Reply:
After offline discusson we found together the naming makes sense. I will resolve this item